crates/contracts/risc0-recursion/tests/recursion.rs

```diff
@@ -1,83 +1,57 @@
-use std::collections::HashMap;
-
-use hyrun::{Cli, Context, ContractData};
+use client_sdk::{
+    contract_states,
+    transaction_builder::{ProvableBlobTx, TxExecutorBuilder},
+};
+use hydentity::{client::register_identity, Hydentity};
 use risc0_recursion::ProofInput;
-use sdk::{HyleOutput, StateDigest};
+use sdk::{ContractName, Digestable, HyleOutput, StateDigest};
 
-#[test_log::test(test)]
-fn test_recursion() {
-    let temp_dir = tempfile::tempdir().unwrap();
+contract_states!(
+    struct States {
+        hydentity: Hydentity,
+    }
+);
 
+#[test_log::test(tokio::test)]
+async fn test_recursion() {
     std::env::set_var("RISC0_DEV_MODE", "1");
 
-    let hardcoded_initial_states = {
-        let mut map = HashMap::new();
-        map.insert(
-            "hydentity".to_owned(),
-            StateDigest(
-                bincode::encode_to_vec(hydentity::Hydentity::new(), bincode::config::standard())
-                    .expect("Failed to encode Hydentity"),
-            ),
-        );
-        map
-    };
-
-    hyrun::run_command(&Context {
-        cli: Cli {
-            command: hyrun::CliCommand::Hydentity {
-                command: hyrun::HydentityArgs::Register {
-                    account: "toto1".to_owned(),
-                },
-            },
-            user: Some("bob".to_owned()),
-            password: Some("bob".to_owned()),
-            nonce: Some(0),
-            host: "".to_owned(),
-            port: 0,
-            proof_path: temp_dir.path().to_str().unwrap().to_owned(),
-        },
-        contract_data: ContractData {
-            hydentity_elf: hyle_contracts::HYDENTITY_ELF.to_vec(),
-            hydentity_id: hyle_contracts::HYDENTITY_ID.to_vec(),
-            ..ContractData::default()
-        },
-        hardcoded_initial_states: hardcoded_initial_states.clone(),
+    let mut executor = TxExecutorBuilder::default().with_state(States {
+        hydentity: StateDigest(
+            bincode::encode_to_vec(hydentity::Hydentity::new(), bincode::config::standard())
+                .expect("Failed to encode Hydentity"),
+        )
+        .try_into()
+        .unwrap(),
```

Comment:
c'est weird ce que tu fais là non ? Pourquoi pas 
```suggestion
        hydentity: Hydentity::new()
```

Reply:
je crois que c'est juste que j'avais pas compris à l'époque